h2o-r/tests/testdir_algos/gam/runit_GH_16149_gam_row_error.R

```diff
@@ -0,0 +1,56 @@
+setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
+source("../../../scripts/h2o-r-test-setup.R")
+
+library(data.table)
+library(dplyr)
+library(h2o) # or load your h2o in a different way
+library(ggplot2)
+
+# This test was provided by a customer.  No exit condition is needed as
+# the test before my fix always failed.  As long as this test completes
+# successfully, it should be good enough.
+test.gam.dataset.error <- function(n) {
+  sum_insured <- seq(1, 200000, length.out = n)
```

Comment:
I wonder what in the tests is crucial that do not change _result and actually break the h2o. I tried it with prostate.csv and it worked so it must be the way how the data are generated.

Reply:
Adam

This is totally crazy!  The dataset that they generated will be rebalanced into two chunks from one chunk after the gamified columns are added.  It is totally coincidence that we ran into this problem.

Other datasets may not have their dataset chunk number changes and that is why this error never occur to them!

Reply:
Waw, good catch!